ShopAPI/ShopAPI.Tests/HTTPResponseTests.cs

```diff
@@ -0,0 +1,47 @@
+
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.VisualStudio.TestPlatform.TestHost;
+using System.Net.Http;
+using ShopAPI.Models;
+using Newtonsoft.Json.Linq;
+
+namespace ShopAPI.Tests
+{
+    public class HTTPResponseTests : IClassFixture<WebApplicationFactory<Program>>
+    {
+        // Arrange
+        readonly HttpClient _client;
+        public HTTPResponseTests(WebApplicationFactory<Program> app)
+        {
+            _client = app.CreateClient();
+        }
+
+        [Fact]
+        public async Task HTTPResponseGetCartTest()
+        {
+            // Act
+            var response = await _client.GetAsync("api/GetCart/1");
+            // Assert
+            response.EnsureSuccessStatusCode();
+        }
+
+        [Fact]
+        public async Task HTTPResponseGetTotalsTest()
+        {
+            // Act
+            var response = await _client.GetAsync("api/GetTotals/1");
+            // Assert
+            response.EnsureSuccessStatusCode();
+        }
+
+        [Fact]
+        public async Task HTTPResponseGetAllItemsTest()
+        {
+            // Act
+            var response = await _client.GetAsync("api/Item/GetAllItems");
+            // Assert
+            response.EnsureSuccessStatusCode();
+        }
```

Comment:
These tests should be independent of what's already in the database. Add an arrange section to each test.

Reply:
So create a new client for each test?

Reply:
No. For example `HTTPResponseGetCartTest` just calls the GetCard method with 1 as the parameter. You need to arrange some some products in a cart, then call the GetCart endpoint. You would also need to ensure the products you added to the cart are a part of that response too. Testing seems a little bare-bone right now

Reply:
I'm confused because I'm just testing the Http response not the functionality of the methods in the database. Im only testing for a 200 or 404 response from the http methods.

Reply:
Thats not enough. The tests fail with an empty database
![image](https://github.com/evan-scales/PalmetoProgrammers547/assets/70162138/fd0aa74c-391c-4a17-b11b-6abf82cb63c2)